### ISSUE TITLE
Scylla Manager crashed and cannot start (could be due to https://github.com/scylladb/scylla-manager/pull/3941 )

### ISSUE THREAD

**Issue by gdubicki** [scylladb/scylla-manager#3972]
We run Scylla Manager 3.3.0 with Scylla 5.4.9. Today I noticed that the manager is crashed and is in CrashLoopBackoff after 24 failed restarts.

The logs:
```
{"L":"INFO","T":"2024-08-15T08:56:33.135Z","N":"backup.index","M":"Done indexing snapshot files","duration":"8.050019781s","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.241.174","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.243.109","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.241.174","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.243.109","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.241.175","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.241.175","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.249.238","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.249.238","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.241.130","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.241.130","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:38.678Z","N":"http","M":"GET /api/v1/clusters","from":"10.6.249.1:39518","status":200,"bytes":700,"duration":"7ms","_trace_id":"9O5dDq_dQAO1BsYAZhDeMQ"}
{"L":"INFO","T":"2024-08-15T08:56:48.677Z","N":"http","M":"GET /api/v1/clusters","from":"10.6.249.1:54724","status":200,"bytes":700,"duration":"6ms","_trace_id":"sx_qPHGWTmy3YGdDCe7dRw"}
{"L":"INFO","T":"2024-08-15T08:56:58.680Z","N":"http","M":"GET /api/v1/clusters","from":"10.6.249.1:36986","status":200,"bytes":700,"duration":"9ms","_trace_id":"J5MXev4KQHGmBMguVD7RGg"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1174b4d]

goroutine 2125 [running]:
github.com/scylladb/scylla-manager/v3/pkg/scyllaclient.(*Client).RcloneDeletePaths(0xc000002000, {0x2d81720, 0xc000371500}, {0xc000ba3130, 0xc}, {0xc000
b2b8c0?, 0xc01515a000?}, {0xc01515d008, 0x2e2, 0x2ff})
    github.com/scylladb/scylla-manager/v3/pkg/scyllaclient/client_rclone.go:384 +0x20d
github.com/scylladb/scylla-manager/v3/pkg/scyllaclient.(*Client).RcloneDeletePathsInBatches(0xc000002000, {0x2d81720, 0xc000371500}, {0xc000ba3130, 0xc}
, {0xc000b2b8c0, 0x5a}, {0xc01515a008, 0x2e2, 0x2e2}, ...)
    github.com/scylladb/scylla-manager/v3/pkg/scyllaclient/client_rclone.go:353 +0x1e7
github.com/scylladb/scylla-manager/v3/pkg/service/backup.(*worker).deduplicateHost.func2(0x0?)
    github.com/scylladb/scylla-manager/v3/pkg/service/backup/worker_deduplicate.go:101 +0x8af
github.com/scylladb/scylla-manager/v3/pkg/util/parallel.Run.func1()
    github.com/scylladb/scylla-manager/v3/pkg/util/parallel/parallel.go:72 +0xb2
created by github.com/scylladb/scylla-manager/v3/pkg/util/parallel.Run in goroutine 1932
    github.com/scylladb/scylla-manager/v3/pkg/util/parallel/parallel.go:57 +0xe8
Stream closed EOF for scylla-manager/scylla-manager-984b44d9c-4m8km (scylla-manager)
```

The cluster repair task was in progress.

**Comment by gdubicki** [comment]
We think that it's related to deduplication of backups. We will try to work around it.

**Comment by mykaul** [comment]
I wonder if it's related to https://github.com/scylladb/scylla-manager/pull/3941 - but that looks like a 3.3.1 item.

**Comment by reneklacan** [comment]
@mykaul we managed to resolve issue but this happened:

1. I ran `nodetool clearsnapshot` on every scylla node to clean up snapshots
2. Then manager crashed on `"db init: database is ahead"` (from https://github.com/scylladb/gocqlx/blob/e7267922c4da8588a6f587aed9f3e821ed4694fb/migrate/migrate.go#L124-L127)
3. After debugging I found out that `SELECT * FROM scylla_manager.gocqlx_migrate;` output contains `v3.3.1.cql`
4. I double checked our manager image and we were always on `scylladb/scylla-manager:3.3.0`
5. I updated image to `scylladb/scylla-manager:3.3.1-0.20240814.5e4c781a2`
6. Manager booted and has been running fine for a couple of minutes

Not sure how it is possible but maybe commit with migration https://github.com/scylladb/scylla-manager/commit/1f1cd6c4d991323a02f81f55738a120df4eb4db9 accidentally slipped into one of 3.3.0 releases/images?

**Comment by mykaul** [comment]
> Not sure how it is possible but maybe commit with migration [1f1cd6c](https://github.com/scylladb/scylla-manager/commit/1f1cd6c4d991323a02f81f55738a120df4eb4db9) accidentally slipped into one of 3.3.0 releases/images?

Yes, we had some mishap in our release pipeline yesterday, so it could have been that.

**Comment by reneklacan** [comment]
So unsurprisingly after attempted backup, the original issue came back and is again crashing our scylla manager.

So https://github.com/scylladb/scylla-manager/pull/3941 definitely introduced this problem here.

**Comment by mykaul** [comment]
@karol-kokoszka , @Michal-Leszczynski - let's fix this and respin 3.3.1.

**Comment by karol-kokoszka** [comment]
The PR is merged to branch-3.3.
But AFAIU, there are still problems with building / promoting manager-3.3.

@Annamikhlin Is everything fine now with the https://jenkins.scylladb.com/view/scylla-manager/job/manager-3.3/ ? Can I trigger the release?

**Comment by Annamikhlin** [comment]
> The PR is merged to branch-3.3. But AFAIU, there are still problems with building / promoting manager-3.3.
> 
> @Annamikhlin Is everything fine now with the https://jenkins.scylladb.com/view/scylla-manager/job/manager-3.3/ ? Can I trigger the release?

Yes, for now as I know the release process is ok. you can trigger the release. 
There is already a [release 3.3.1 document](https://docs.google.com/document/d/1WzQM0FOOXt6u-q3wXTlBDGwSanzraKoPQO4NmfqlWbo/) but probably you need to respin the version

**Comment by karol-kokoszka** [comment]
Thanks @Annamikhlin . I will proceed with the release. But need to validate latest run of dtests.
